docs/spec.md

```diff
@@ -699,8 +699,9 @@ it only exists to establish data dependencies from `result` to `inputs`.
 #### Semantics
 
 Within each process group in the StableHLO process grid, concatenates the values
-of the `operand` tensor from each process along `all_gather_dim` and produces a
-`result` tensor.
+of the `operands` tensors from each process along `all_gather_dim` and produces a
+`results` tensors. When multiple `operands` are specified, the `all_gather` is
+performed on each operand.
```

Comment:
And I'm assuming a result tensor is produced for each operand? Consider mentioning that explicitly if so. (Same comment below.)

Reply:
end up removing this part, because `operands..` and `results..` by convention tell the same. I see other specs usually don't explicitly describe variadic inputs behavior.